doc/api/test.md

```diff
@@ -1044,6 +1044,134 @@ exports[`suite of snapshot tests > snapshot test 2`] = `
 Once the snapshot file is created, run the tests again without the
 `--test-update-snapshots` flag. The tests should pass now.
 
+### Inline snapshots
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+Inline snapshots store the expected snapshot content directly within the test
+file as a template literal, rather than in an external `.snapshot` file. This
+makes it easier to see what the test expects and reduces the number of files in
+your project. Inline snapshots are particularly useful for small snapshots
+where having the expected value close to the test provides better readability.
+
+The `t.assert.inlineSnapshot()` function creates inline snapshot assertions.
+When called without a second argument, the function will insert the serialized
+value as a template literal directly into your source code when run with the
+[`--test-update-snapshots`][] flag.
+
+```mjs
+import { test } from 'node:test';
+
+test('inline snapshot test', (t) => {
+  const user = { name: 'Alice', age: 30 };
+
+  // Initially call without second argument
+  t.assert.inlineSnapshot(user);
```

Comment:
Great name! I think this aligns really well to the current TestContext.assert.snapshot() API. Also, I'm pleasantly surprised I don't need to await here.